src/dali/plugin/pair_converter/ccxt_converter.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2022 macanudo527
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import re
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, NamedTuple, Optional, Set, Union
+
+from ccxt import binance
+from rp2.logger import create_logger
+from rp2.rp2_decimal import RP2Decimal
+
+from dali.abstract_pair_converter_plugin import AbstractPairConverterPlugin
+from dali.historical_bar import HistoricalBar
+
+# Native format keywords
+_ID: str = "id"
+
+# Time in ms
+_MINUTE: str = "1m"
+_FIVEMINUTE: str = "5m"
+_FIFTEENMINUTE: str = "15m"
+_ONEHOUR: str = "1h"
+_FOURHOUR: str = "4h"
+_ONEDAY: str = "1d"
+_TIMEGRANULARITY: List[str] = [_MINUTE, _FIVEMINUTE, _FIFTEENMINUTE, _ONEHOUR, _FOURHOUR, _ONEDAY]
+_TIMEGRANULARITY_IN_S: List[int] = [60, 300, 900, 3600, 14400, 86400]
+
+# Currently supported exchanges
+_BINANCE: str = "Binance.com"
+_EXCHANGEDICT: Dict[str, Any] = {_BINANCE: binance}
+
+# Time constants
+_MS_IN_SECOND: int = 1000
+
+
+class StableCoin(NamedTuple):
+    symbol: str
+    fiat_code: str
+
+
+# Default Stable coin for each exchange
+# This should be the stable coin with the most volume on the exchange
+_DEFAULTSTABLEDICT = {
+    _BINANCE: StableCoin(symbol="USDT", fiat_code="USD"),
+}
+
+
+class PairConverterPlugin(AbstractPairConverterPlugin):
+
+    # pricing_alt allows for the user to config specific markets to be used to price an asset that doesn't have
+    # a stable coin or fiat market
+    def __init__(self, historical_price_type: str, pricing_alt: Optional[str] = None) -> None:
+        super().__init__(historical_price_type=historical_price_type)
+        self.__logger: logging.Logger = create_logger(f"CCXT converter using {historical_price_type}")

```

Comment:
Nit: avoid spaces in logger name. Something like:
```suggestion
        self.__logger: logging.Logger = create_logger(f"{self.name()}/{historical_price_type}")
```